pail/interface.go

```diff
@@ -123,6 +121,7 @@ type bucketItemImpl struct {
 }
 
 func (bi *bucketItemImpl) Name() string   { return bi.key }
+func (bi *bucketItemImpl) Hash() string   { return bi.hash }
```

Comment:
What populates hash values for local and gridfs impls?

Reply:
They aren't using `hash`, I added it for s3 in order to avoid making two separate API calls since we get the `ETag` for free when we iterate with the List function. The hashes are calculated using `md5sum` and `sha1sum` helper functions you created.